sycl/test-e2e/ESIMD/unified_memory_api/Inputs/block_store.hpp

```diff
@@ -256,26 +255,48 @@ bool testSLM(queue Q, uint32_t Groups, StorePropertiesT StoreProperties) {
          uint32_t ElemOff = GlobalID * N;
          simd<T, N> Vals(ElemOff, 1);
 
+         if (LocalID == 0) {
+           for (int I = 0; I < Size; I++) {
+             simd<T, 1> InVec(OutVal);
+             slm_block_store(I * sizeof(T), InVec);
+           }
+         }
+         barrier();
+
          if constexpr (UseMask) {
            simd_mask<1> Mask = (GlobalID + 1) & 0x1;
-           simd<T, N> PassThru = Out_val;
-           if constexpr (CheckProperties)
-             slm_block_store(LocalElemOffset, Vals, Mask, StorePropertiesT{});
-           else
-             slm_block_store(LocalElemOffset, Vals, Mask);
-           Vals = slm_block_load<T, N>(LocalElemOffset, Mask, PassThru);
            Vals += 6;
-           slm_block_store(LocalElemOffset, Vals, Mask);
-           Vals = slm_block_load<T, N>(LocalElemOffset, Mask, PassThru);
+           if constexpr (CheckProperties) {
+             if (GlobalID % 2 == 0)
+               slm_block_store<T, N>(LocalElemOffset, Vals, Mask,
+                                     StorePropertiesT{});
+             else
+               slm_block_store<T, N>(LocalElemOffset,
+                                     Vals.template select<N, 1>(), Mask,
+                                     StorePropertiesT{});
+           }
+
+           else
+             slm_block_store<T, N>(LocalElemOffset, Vals, Mask);
+
+           Vals = slm_block_load<T, N>(LocalElemOffset);
+
          } else {
            if constexpr (CheckProperties)
-             slm_block_store(LocalElemOffset, Vals, StorePropertiesT{});
+             if (GlobalID % 2 == 0)
+
+               slm_block_store<T, N>(LocalElemOffset, Vals, StorePropertiesT{});
+             else
+               slm_block_store<T, N>(LocalElemOffset,
+                                     Vals.template select<N, 1>(),
+                                     StorePropertiesT{});
+
            else
-             slm_block_store(LocalElemOffset, Vals);
+             slm_block_store<T, N>(LocalElemOffset, Vals);
```

Comment:
nit: IMO, it would be better to not pass `<T, N>`, which would make the test closer to what user is supposed to do and make the call less verbose and make the check stronger.

Reply:
did this for all cases besides simd_view, we need to specify there because the compiler can't autodeduce, and i double checked to make sure the template parameters are the same as existing apis, so i think this is just a simd_view thing